apps/src/templates/teacherDashboard/SectionActionBox.jsx

```diff
@@ -0,0 +1,216 @@
+    const courseId = this.props.sectionCode.replace(/^[GC]-/, '');
+    this.props.updateRoster(courseId, this.props.sectionName)
+      .then(() => {
+        // While we are embedded in an angular page, reloading is the easiest
+        // way to pick up roster changes.  Once everything is React maybe we
+        // won't need to do this.
+        utils.reload();
+      });
+  };
+
+  onRequestDelete = () => {
+    this.setState({deleting: true});
+  };
+
+  render() {
+    const {sectionData} = this.props;
+    return (
+      <div style={{...styles.actionBox, ...this.props.style}}>
+        <a href={pegasus('/teacher-dashboard#/sections/' + sectionData.id)}>
+          <div style={styles.actionText}>
+            {i18n.sectionViewProgress()}
+          </div>
+        </a>
+        <a href={pegasus('/teacher-dashboard#/sections/' + sectionData.id + "/manage")}>
+          <div style={styles.actionText}>
+            {i18n.manageStudents()}
+          </div>
+        </a>
+        <a href={pegasus('/teacher-dashboard#/sections/' + sectionData.id + '/print_signin_cards')}>
+          <div style={styles.actionText}>
+            {i18n.printLoginCards()}
+          </div>
+        </a>
+        <a>
+          <div style={styles.actionTextBreak} onClick={this.onClickEdit} id="ui-test-edit-section">
+            {i18n.editSectionDetails()}
+          </div>
+        </a>
+        <a>
+          <div style={styles.actionText}>
+            <PrintCertificates
+              sectionId={sectionData.id}
+              assignmentName={sectionData.assignmentNames[0]}
+            />
+          </div>
+        </a>
+        {sectionData.loginType === OAuthSectionTypes.clever &&
+          <a>
+            <div style={styles.actionText} onClick={this.onClickSync}>
+              {i18n.syncClever()}
+            </div>
+          </a>
+        }
+        {sectionData.loginType === OAuthSectionTypes.google_classroom &&
+          <a>
+            <div style={styles.actionText} onClick={this.onClickSync}>
+              {i18n.syncGoogleClassroom()}
+            </div>
+          </a>
+        }
+        <a>
+          <div style={styles.actionText} onClick={this.onClickHideShow} id="ui-test-hide-section">
+            {sectionData.hidden ? i18n.showSection() : i18n.hideSection()}
+          </div>
+        </a>
+        {sectionData.studentCount === 0 && (
+          <div>
+            <a>
+              <div style={styles.archiveDelete} onClick={this.onRequestDelete}>
+                <FontAwesome icon=" fa-times-circle" style={styles.xIcon}/>
+                {i18n.deleteSection()}
+              </div>
+            </a>
+            <BaseDialog
+              useUpdatedStyles
+              uncloseable
+              isOpen = {this.state.deleting}
+              assetUrl={() => ''}
+              style={{padding:20}}
+            >
+              <h2>{i18n.deleteSection()}</h2>
+              <div>{i18n.deleteSectionConfirm()}</div>
```

Comment:
This dialog should include the section name somewhere in its title or description text - otherwise we're depending on the user remembering which section they picked before the dialog opened, which isn't obvious after the fact.

Reply:
Also note that we're making a dialog for every section row.  For users with a _ton_ of sections it may work better to render the dialog once at the page level and show/hide it based on redux state.  After all, you shouldn't be able to show more than one delete dialog at a time.